src/models/project/project.spec.ts

```diff
@@ -0,0 +1,288 @@
+          endDate: $endDate
+          estimeeSpentTime: $estimeeSpentTime
+        ) {
+          id
+          status
+          startDate
+          endDate
+          estimeeSpentTime
+        }
+      }
+    `;
+
+    const variables = {
+      status: 'DONE',
+      startDate: '2021-02-07T21:04:39.573Z',
+      endDate: '2021-02-07T21:04:39.573Z',
+      estimeeSpentTime: 5.0,
+    };
+
+    const response = await server.executeOperation({
+      query: addProjectMutation,
+      variables,
+    });
+
+    id = response.data?.createProject.id;
+    expect(response.data?.createProject.id).toEqual(id);
+    expect(response.data?.createProject).toHaveProperty('status', variables.status);
+    expect(response.data?.createProject).toHaveProperty('startDate', '1612731879573');
+    expect(response.data?.createProject).toHaveProperty('endDate', '1612731879573');
+    expect(response.data?.createProject).toHaveProperty('estimeeSpentTime', variables.estimeeSpentTime);
+  });
+
+  // * GET ALL PROJECTS TEST
+  it('should get all projects', async () => {
+    const query = gql`
+    query Query {
+        getAllProjects {
+          id
+          status
+          members {
+            userId
+            projectId
+            projectRole
+          }
+          tasks {
+            id
+            subject
+            projectId
+            users {
+              lastName
+              id
+              email
+              password
+              firstName
+              avatar
+              createdAt
+              updatedAt
+            }
+            startDate
+            endDate
+            estimeeSpentTime
+            advancement
+            comments {
+              id
+              text
+              userId
+              taskId
+            }
+            tags {
+              id
+              label
+              color
+              tasks {
+                id
+                subject
+                projectId
+                users {
+                  id
+                  email
+                  password
+                  firstName
+                  lastName
+                  avatar
+                  createdAt
+                  updatedAt
+                }
+                startDate
+                endDate
+                estimeeSpentTime
+                advancement
+                comments {
+                  id
+                  text
+                  userId
+                  taskId
+                }
+                tags {
+                  id
+                  label
+                  color
+                  tasks {
+                    id
+                    subject
+                    projectId
+                    users {
+                      id
+                      email
+                      password
+                      firstName
+                      lastName
+                      avatar
+                      createdAt
+                      updatedAt
+                    }
```

Comment:
😱 Je crois qu'on appelle ça un Wall of Doom dans le milieu xD
En faisant ça, tu vas query presque toute la BDD en une seule requête, quand on aura des données en BDD le test sera en PLS x)

Tu peux juste query les IDs, pour le test c'est amplement suffisant, on cherche juste à savoir si la requête renvoie bien un tableau d’éléments 🙂